scribejava-apis/src/main/java/com/github/scribejava/ProductHuntApi.java

```diff
@@ -0,0 +1,50 @@
+package com.github.scribejava.apis;
+
+import com.github.scribejava.apis.service.ProductHuntServiceImpl;
+import com.github.scribejava.core.builder.api.DefaultApi20;
+import com.github.scribejava.core.extractors.AccessTokenExtractor;
+import com.github.scribejava.core.extractors.JsonTokenExtractor;
+import com.github.scribejava.core.model.OAuthConfig;
+import com.github.scribejava.core.model.OAuthConstants;
+import com.github.scribejava.core.model.Verb;
+import com.github.scribejava.core.oauth.OAuthService;
+import com.github.scribejava.core.utils.OAuthEncoder;
+
+public class ProductHuntApi extends DefaultApi20 {
+    private static final String AUTHORIZE_URL = "https://api.producthunt.com/v1/oauth/authorize?client_id=%s&redirect_uri=%s&response_type=code&scope=public+private";
+    private static final String ACCESS_TOKEN_URL = "https://api.producthunt.com/v1/oauth/token?grant_type=authorization_code";
+
+    @Override
+    public Verb getAccessTokenVerb() {
+        return Verb.POST;
+    }
+
+    @Override
+    public String getAccessTokenEndpoint() {
+        return ACCESS_TOKEN_URL;
+    }
+
+    @Override
+    public String getAuthorizationUrl(final OAuthConfig config) {
+        Preconditions.checkValidUrl(config.getCallback(),
+                "Must provide a valid url as callback. Product Hunt does not support OOB");
+        final StringBuilder sb = new StringBuilder(String.format(AUTHORIZE_URL, config.getApiKey(), OAuthEncoder.encode(
+                config.getCallback())));
+        if (config.hasScope()) {
+            sb.append('&').append(OAuthConstants.SCOPE).append('=').append(OAuthEncoder.encode(config.getScope()));
```

Comment:
I see, that scope is parametrized actually ;-)
"&scope=public+private" looks like redundant part of AUTHORIZE_URL